vector-stores/spring-ai-pgvector-store/src/main/java/org/springframework/ai/vectorstore/PgVectorStore.java

```diff
@@ -535,12 +558,35 @@ public Builder withInitializeSchema(boolean initializeSchema) {
 			return this;
 		}
 
+		public Builder withObservationRegistry(ObservationRegistry observationRegistry) {
+			this.observationRegistry = observationRegistry;
+			return this;
+		}
+
+		public Builder withSearchObservationConvention(
+				VectorStoreObservationConvention customSearchObservationConvention) {
+			this.searchObservationConvention = customSearchObservationConvention;
+			return this;
+		}
+
 		public PgVectorStore build() {
 			return new PgVectorStore(schemaName, vectorTableName, vectorTableValidationsEnabled, jdbcTemplate,
 					embeddingModel, dimensions, distanceType, removeExistingVectorStoreTable, indexType,
-					initializeSchema);
+					initializeSchema, observationRegistry, searchObservationConvention);
 		}
 
 	}
 
+	@Override
+	public VectorStoreObservationContext.Builder createObservationContextBuilder(String operationName) {
+
+		return VectorStoreObservationContext.builder("pg_vector", operationName)
```

Comment:
Consider introducing a `VectorStoreProvider` enum in the same package as `AiProvider`, and have the "pg_vector" value as part of that enum (but still leaving the field type as String, similar to how we handle the model provider in the LLM observations).

Reply:
I will create an issue to consider this change once this PR is merged.

Reply:
Added entry for the Simple Vector Store as well.